extendr-api/tests/rng_tests.rs

```diff
@@ -0,0 +1,40 @@
+//! Tests the ability to use R to generate random numbers.
+//!
+//! The `extendr` attribute macro now has a `use_rng` argument.
+//! If this is used, then a `GetRNGstate()` is invoked before the function is called,
+//! and a `PutRNGstate()` is invoked after, even if said method panics.
+//!
+//! These tests has to check two things:
+//!
+//! * That the seed is set and affects the sampling functions on the rust-side
+//! in the same way, as on the r-side.
+//! * That the resulting sampled vector is not equal 0, as it would be that
+//! if `GetRNGstate()` is not called prior to sampling.
+//!
+//!
+use extendr_api::prelude::*;
+
+#[extendr(use_rng = true)]
+fn generate_big_random_vec() -> Vec<f64> {
+    let n = 100;
+    let param = 10e3;
+    (0..n)
+        .map(|_| unsafe { libR_sys::R_unif_index(param) })
+        .collect()
+}
+
+#[test]
+fn test_extendr_rng() {
+    test! {
+        R!(r#"set.seed(20230205)"#).unwrap();
+        let x = generate_big_random_vec();
+        R!(r#"set.seed(20230205)"#).unwrap();
+        let y = generate_big_random_vec();
+        assert_eq!(y, x);
```

Comment:
Just to be safe, does `assert_eq!` correctly compare iterable types like `Vec<T>`?

Reply:
Sure:
```rust
>> assert_eq!(vec![13,4,3], vec![22]);
thread '<unnamed>' panicked at 'assertion failed: `(left == right)`
  left: `[13, 4, 3]`,
 right: `[22]`', src/lib.rs:6:1
stack backtrace:
   0: rust_begin_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:67:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed
   4: std::panicking::try
   5: run_user_code_1
   6: evcxr::runtime::Runtime::run_loop
   7: evcxr::runtime::runtime_hook
   8: evcxr::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
>> assert_eq!(vec![13,4,3], vec![13,4,3]);
```